File: frontend/src/types/checklist.ts
export interface ChecklistItem {
  id: string;
  category: string;  // "즉시 조치" | "법적 의무" | "금지 사항"
  item: string;
  description?: string;
  priority: number;
  is_mandatory: boolean;
  source_type?: string;  // "gpt" | "norm_obligation" | "norm_prohibition"
  source_ref?: string;   // "제42조" 등 조항번호
}

export interface Checklist {
  title: string;
  workplace_type?: string;
  items: ChecklistItem[];
}

File: backend/app/models/checklist.py
from pydantic import BaseModel
from typing import List, Optional


class ChecklistItem(BaseModel):
    id: str
    category: str
    item: str
    description: Optional[str] = None
    priority: int
    is_mandatory: bool


class Checklist(BaseModel):
    title: str
    workplace_type: Optional[str] = None
    items: List[ChecklistItem]
